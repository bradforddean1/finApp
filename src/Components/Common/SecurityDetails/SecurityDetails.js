import React from "react";
import PropTypes from "prop-types";
import World from "../../Common/icons/WorldIcon/WorldIcon";
import "./SecurityDetails.scss";

/**
 * Footer of the header card with basic security summary.
 * @component
 */
function SecurityDetails(props) {
	return (
		<div className="SecurityDetails card-line">
			<div className="web">
				{!!props.website && (
					<>
						<World />
						<a className="small-script" href={props.website}>
							{props.website}
						</a>
					</>
				)}
			</div>
			<span className="small-script">{props.country}</span>
			<span className="small-script">IPO: {props.ipo}</span>
		</div>
	);
}

SecurityDetails.propTypes = {
	/**
	 * Displayable date of IPO
	 */
	ipo: PropTypes.string,
	/**
	 * Displayable headquarters
	 */
	country: PropTypes.string,
	/**
	 * Displayable company wenbsite
	 */
	website: PropTypes.string,
};

SecurityDetails.defaultProps = {
	ipo: "-",
	country: "-",
	website: null,
};

export default SecurityDetails;
