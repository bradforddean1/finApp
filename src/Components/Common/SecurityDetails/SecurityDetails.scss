.SecurityDetails {
	display: flex;
	justify-content: left;
	align-items: center;
	flex-wrap: wrap;
	font-weight: 500;
	font-size: 0.6667em;
	line-height: 0.6667em;
}

.SecurityDetails .web {
	display: flex;
	align-items: center;
}

.SecurityDetails > * {
	margin-right: 1.89em;
	margin-bottom: 1em;
}
.SecurityDetails > span {
	margin-left: 0.3em;
}

.SecurityDetails i {
	height: 1.33em;
	width: 1.33em;
}
