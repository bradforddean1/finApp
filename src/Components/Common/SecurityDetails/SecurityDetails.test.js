import React from "react";
import ReactDOM from "react-dom";
import SecurityDetails from "./SecurityDetails";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("SecurityDetails Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(<SecurityDetails />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<SecurityDetails />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	const props = { ipo: "2000-01-10", website: "www.xyz.com", Country: "AUS" };

	it("Renders given props", () => {
		const wrapper = shallow(<SecurityDetails {...props} />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
