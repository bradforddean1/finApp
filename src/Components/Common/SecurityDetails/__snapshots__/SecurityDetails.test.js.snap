// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityDetails Component Renders given props 1`] = `
<div
  className="SecurityDetails card-line"
>
  <div
    className="web"
  >
    <WorldIcon />
    <a
      className="small-script"
      href="www.xyz.com"
    >
      www.xyz.com
    </a>
  </div>
  <span
    className="small-script"
  >
    -
  </span>
  <span
    className="small-script"
  >
    IPO: 
    2000-01-10
  </span>
</div>
`;

exports[`SecurityDetails Component Renders the Default State 1`] = `
<div
  className="SecurityDetails card-line"
>
  <div
    className="web"
  />
  <span
    className="small-script"
  >
    -
  </span>
  <span
    className="small-script"
  >
    IPO: 
    -
  </span>
</div>
`;
